cmd/podman/farm/build.go

```diff
@@ -0,0 +1,128 @@
+package farm
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/containers/common/pkg/config"
+	"github.com/containers/podman/v4/cmd/podman/common"
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/containers/podman/v4/cmd/podman/utils"
+	"github.com/containers/podman/v4/pkg/farm"
+	"github.com/containers/storage"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type buildOptions struct {
+	buildOptions common.BuildFlagsWrapper
+	local        bool
+	platforms    []string
+}
+
+var (
+	farmBuildDescription = `Build images on farm nodes, then bundle them into a manifest list`
+	buildCommand         = &cobra.Command{
+		Use:     "build",
+		Short:   "Build a container image for multiple architectures",
+		Long:    farmBuildDescription,
+		RunE:    build,
+		Example: "podman farm build [flags] buildContextDirectory",
+		Args:    cobra.ExactArgs(1),
+	}
+	buildOpts = buildOptions{
+		buildOptions: common.BuildFlagsWrapper{},
+	}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: buildCommand,
+		Parent:  farmCmd,
+	})
+	flags := buildCommand.Flags()
+	flags.SetNormalizeFunc(utils.AliasFlags)
+
+	localFlagName := "local"
+	// Default for local is true and hide this flag for the remote use case
+	if !registry.IsRemote() {
+		flags.BoolVarP(&buildOpts.local, localFlagName, "l", true, "Build image on local machine as well as on farm nodes")
+	}
+	cleanupFlag := "cleanup"
+	flags.BoolVar(&buildOpts.buildOptions.Cleanup, cleanupFlag, false, "Remove built images from farm nodes on success")
+	platformsFlag := "platforms"
+	buildCommand.PersistentFlags().StringSliceVar(&buildOpts.platforms, platformsFlag, nil, "Build only on farm nodes that match the given platforms")
+
+	common.DefineBuildFlags(buildCommand, &buildOpts.buildOptions, true)
+}
+
+func build(cmd *cobra.Command, args []string) error {
+	opts, err := common.ParseBuildOpts(cmd, args, &buildOpts.buildOptions)
+	if err != nil {
+		return err
+	}
+	// Close the logFile if one was created based on the flag
+	if opts.LogFileToClose != nil {
+		defer opts.LogFileToClose.Close()
+	}
+	if opts.TmpDirToClose != "" {
+		// We had to download the context directory.
+		// Delete it later.
+		defer func() {
+			if err = os.RemoveAll(opts.TmpDirToClose); err != nil {
+				logrus.Errorf("Removing temporary directory %q: %v", opts.ContextDirectory, err)
```

Comment:
Is the directory named in the error message supposed to be different from the one passed to `os.RemoveAll()`?

Reply:
I was wondering the same thing when I copied it from `podman build` but fixed to show the actual directory name on error